File: dev_server.py
#!/usr/bin/env python3
"""Minimal static dev server so `npm run dev` works for this static site.

Kimi Work preview may pass --host/--port (or set HOST/PORT env); a bare
positional port is also accepted, mirroring `python3 -m http.server`.
Stdlib only — works with any Python 3.
"""
import argparse
import os
from http.server import SimpleHTTPRequestHandler, ThreadingHTTPServer


def main() -> None:
    parser = argparse.ArgumentParser()
    parser.add_argument("--port", type=int, default=int(os.environ.get("PORT", "7100")))
    parser.add_argument("--host", default=os.environ.get("HOST", "127.0.0.1"))
    parser.add_argument("positional_port", nargs="?", type=int, default=None)
    args, _unknown = parser.parse_known_args()

    port = args.positional_port or args.port
    os.chdir(os.path.dirname(os.path.abspath(__file__)))
    print(f"Serving {os.getcwd()} at http://{args.host}:{port}", flush=True)
    server = ThreadingHTTPServer((args.host, port), SimpleHTTPRequestHandler)
    try:
        server.serve_forever()
    except KeyboardInterrupt:
        pass


if __name__ == "__main__":
    main()

File: dev_server.js
#!/usr/bin/env node
/* Minimal static dev server so `npm run dev` works for this static site.
 *
 * Kimi Work preview may pass --host/--port (or set HOST/PORT env); a bare
 * positional port is also accepted, mirroring `python3 -m http.server`.
 * Zero dependencies — works with any Node 18+.
 */
const http = require("http");
const fs = require("fs");
const path = require("path");

const args = process.argv.slice(2);
function argValue(flag) {
  const i = args.indexOf(flag);
  return i !== -1 && args[i + 1] ? args[i + 1] : null;
}
const positional = args.find((a) => /^\d+$/.test(a));
const host =
  argValue("--host") || argValue("-h") || process.env.HOST || "127.0.0.1";
const port = Number(
  positional || argValue("--port") || argValue("-p") || process.env.PORT || 7100
);

const root = __dirname;
const MIME = {
  ".html": "text/html; charset=utf-8",
  ".css": "text/css; charset=utf-8",
  ".js": "text/javascript; charset=utf-8",
  ".json": "application/json; charset=utf-8",
  ".png": "image/png",
  ".jpg": "image/jpeg",
  ".jpeg": "image/jpeg",
  ".gif": "image/gif",
  ".svg": "image/svg+xml",
  ".webp": "image/webp",
  ".ico": "image/x-icon",
  ".woff": "font/woff",
  ".woff2": "font/woff2",
  ".mp4": "video/mp4",
  ".md": "text/markdown; charset=utf-8",
  ".txt": "text/plain; charset=utf-8",
  ".pdf": "application/pdf",
};

const server = http.createServer((req, res) => {
  let urlPath = decodeURIComponent(req.url.split("?")[0]);
  if (urlPath.endsWith("/")) urlPath += "index.html";
  const file = path.normalize(path.join(root, urlPath));
  if (!file.startsWith(root)) {
    res.writeHead(403).end("Forbidden");
    return;
  }
  fs.readFile(file, (err, data) => {
    if (err) {
      res.writeHead(404).end("Not Found: " + urlPath);
      return;
    }
    res.writeHead(200, {
      "Content-Type": MIME[path.extname(file).toLowerCase()] || "application/octet-stream",
    });
    res.end(data);
  });
});

server.listen(port, host, () => {
  console.log(`Serving ${root} at http://${host}:${port}`);
});
